Bug 2 - Output format update

zachomedia opened the pull request:
Updated output format:
- Removed &nbsp
- Removed extra spaces
- Removed HTML tages
- Removed headers (Notes: and Reserve:)

Date format updated:

```
[dates] => Array
 (
  [start_time] => 07:00
  [end_time] => 09:50
  [weekdays] => T
  [start_date] => 09/17
  [end_date] => 09/17
 )
```

I have also updated the documentation (Readme) to reflect the latest output changes.

zachomedia committed:
Added data beautification.

- Removes &nbsp
- Removes additional spaces
- Removes HTML tages
- Removes headers from select fields.

zachomedia committed:
Formats the dates field.

zachomedia committed:
Documentation update

KartikTalwar commented:
Hey @zachomedia thanks for sanitizing the output!

This pretty much handles the dates and times nicely except for online courses and TBA classes.

``` php
print_r(parse_schedule('1131', 'PHYS', '380'));
print_r(parse_schedule('1139', 'PHYS', '490'));
```
- For some online courses, the dates show up as **12:00-12:00** with no day. See code comments.
- For [PHYS380](http://www.adm.uwaterloo.ca/cgi-bin/cgiwrap/infocour/salook.pl?sess=1131&subject=PHYS&cournum=380), the classes array has length 2
- We should probably add another boolean field to dates, `is_tba`. Thoughts?

KartikTalwar commented on Schedule/schedule.php:
```diff
@@ -140,6 +160,65 @@ function parse_schedule($term_id, $faculty, $course, $level = 'under')
+  $match = array();
+  
+  $strDate = beautify($strDate);
+  $dateRegex = "/(\d{2}:\d{2})-(\d{2}:\d{2})(\w+)\s*(?:(\d{2}\/\d{2})-(\d{2}\/\d{2}))?.*/";
```
Having an optional day matcher we can handle the 12:00-12:00 case by by doing `count($match) >= 3`

``` php
$dateRegex = "/(\d{2}:\d{2})-(\d{2}:\d{2})(\w+)?\s*(?:(\d{2}\/\d{2})-(\d{2}\/\d{2}))?.*/";
```

KartikTalwar commented on Schedule/schedule.php:
```diff
@@ -140,6 +160,65 @@ function parse_schedule($term_id, $faculty, $course, $level = 'under')
+  $strDate = beautify($strDate);
+  $dateRegex = "/(\d{2}:\d{2})-(\d{2}:\d{2})(\w+)\s*(?:(\d{2}\/\d{2})-(\d{2}\/\d{2}))?.*/";
+  $matchResult = preg_match($dateRegex, $strDate, $match);
+      
```
TBA check?

``` php
$tbaRegex = preg_match("/TBA/", $strDate, $tbamatches);

$date['tba'] = count($tbamatches) == 1;
```

KartikTalwar replied:
On the same note, we can also handle **Cancelled Section**

http://www.adm.uwaterloo.ca/cgi-bin/cgiwrap/infocour/salook.pl?sess=1139&subject=PHYS&cournum=771&level=grad

zachomedia committed:
Fixes: Failed date matches and extra classes listed.

1. Fixed cases where dates would fail because there were no weekdays.
2. Fixed cases where extra classes would appear.

zachomedia committed:
Added TBA support for dates.

zachomedia committed:
Added support for cancelled dates.

zachomedia committed:
Renamed boolean fields.

zachomedia committed:
Updated documentation.

zachomedia commented:
I've made a few updates (from the suggestions above).
- Online courses where times are 12:00-12:00, with no weekday, now appear correct.
- I've corrected the number of classes in the array (this was caused the updates to date output)
- I've added support for TBA and Cancelled classes:

```
[dates] => Array
  (
    [start_time] => 08:30
    [end_time] => 10:20
    [weekdays] => F
    [start_date] =>
    [end_date] =>
    [is_tba] =>
    [is_cancelled] =>
  )
```

KartikTalwar commented:
Closes #2
